pkg/controller/kieapp/shared/utils.go

```diff
@@ -155,13 +155,13 @@ func GeneratePassword(length int) []byte {
 	return buf
 }
 
-func isEnvVarInList(envName string, Env []corev1.EnvVar) bool {
-	for _, v := range Env {
+func getEnvVar(envName string, Env []corev1.EnvVar) (int, *corev1.EnvVar) {
+	for pos, v := range Env {
 		if v.Name == envName {
-			return true
+			return pos, &v
```

Comment:
I don't want to return a copy. I prefer to return the pointer in case this value has to be changed which couldn't be achieved by returning a copy. I don't think it should lead into a bug if the developer understands what returning a pointer mean.
Regarding returning only the index, I considered so and I didn't have any strong feeling against any choice, just found this one more flexible.

Reply:
But that's my point. You're returning a pointer to a copy! :-)

Create a unit test and give it a try!

Reply:
Indeed, the function is receiving an array of copies. Thanks for the correction. Shouldn't we be passing arrays of pointers to avoid creating too many copies?